src/main/java/org/skife/jdbi/v2/PreparedBatch.java

```diff
@@ -118,7 +134,12 @@ public PreparedBatch define(final Map<String, ? extends Object> values)
         return executeAndGenerateKeys(new SingleColumnMapper<GeneratedKeyType>(mapper));
     }
 
-    private <Result> Object internalBatchExecute(QueryResultMunger<Result> munger) {
+    public <GeneratedKeyType> GeneratedKeys<GeneratedKeyType> executeAndGenerateKeys(ResultColumnMapper<GeneratedKeyType> mapper,
```

Comment:
I think it might. For example, one could want to get a generated id of a record along with the timestamp it was inserted (the timestamp is generated by the database). The JDBI spec does not prohibit such scenario and I don't see reasons why we should do this.